redwood-treehouse-lazylayout-uiview/src/commonMain/kotlin/app/cash/redwood/treehouse/lazylayout/uiview/UIViewLazyList.kt

```diff
@@ -13,97 +13,141 @@
+      dataSource = tableViewDataSource
+      delegate = tableViewDelegate
+      rowHeight = 64.0 // TODO: size rows by their content.
+      prefetchingEnabled = false
+      registerClass(
+        Cell(UITableViewCellStyleDefault, reuseIdentifier).classForCoder() as ObjCClass?,
+        forCellReuseIdentifier = reuseIdentifier,
+      )
+    }
+
+  private lateinit var onPositionDisplayed: (Int) -> Unit
+
   override fun isVertical(isVertical: Boolean) {
     if (!isVertical) {
       // TODO UITableView only supports vertical scrolling. Switch to UICollectionView.
       TODO()
     }
   }
 
-  override fun intervals(intervals: List<LazyListInterval>) {
-    dataSource.intervals = intervals
-    root.reloadData()
+  override fun onPositionDisplayed(onPositionDisplayed: (Int) -> Unit) {
+    this.onPositionDisplayed = onPositionDisplayed
   }
 
   override var layoutModifiers: LayoutModifier = LayoutModifier
 
-  override val value: UIView get() = root
+  override val value: UIView get() = tableView
 }
 
-private class TableViewDataSource<A : AppService>(
-  private val treehouseApp: TreehouseApp<A>,
-  private val widgetSystem: WidgetSystem,
-) : UITableViewDiffableDataSource() {
-  var intervals = emptyList<LazyListInterval>()
-
-  override fun numberOfSectionsInTableView(tableView: UITableView): NSInteger {
-    return intervals.size.toLong()
-  }
+private class Cell(
+  style: UITableViewCellStyle,
+  reuseIdentifier: String?,
+) : UITableViewCell(style, reuseIdentifier) {
+  private var widget: UIView? = null
 
-  override fun tableView(tableView: UITableView, numberOfRowsInSection: NSInteger): NSInteger {
-    return intervals[numberOfRowsInSection.toInt()].keys.size.toLong()
-  }
+  /** Factory function for a new cell. */
+  override fun initWithStyle(
+    style: UITableViewCellStyle,
+    reuseIdentifier: String?,
+  ): UITableViewCell = Cell(style, reuseIdentifier)
 
-  override fun tableView(tableView: UITableView, cellForRowAtIndexPath: NSIndexPath): UITableViewCell {
-    val treehouseView = TreehouseUIKitView(widgetSystem)
-    val cellContentSource = CellContentSource<A>(
-      intervals[cellForRowAtIndexPath.section.toInt()].itemProvider,
-      cellForRowAtIndexPath.row.toInt(),
-    )
-    cellContentSource.bindWhenReady(treehouseView, treehouseApp)
-    return TableViewCell(treehouseView.view)
+  override fun prepareForReuse() {
+    super.prepareForReuse()
+    this.widget?.removeFromSuperview()
   }
-}
-
-private class CellContentSource<A : AppService>(
-  private val itemProvider: LazyListInterval.Item,
-  private val index: Int,
-) : TreehouseContentSource<A> {
-
-  override fun get(app: A): ZiplineTreehouseUi {
-    return itemProvider.get(index)
-  }
-}
 
-private class TableViewCell(private val view: UIView) :
-  UITableViewCell(UITableViewCellStyle.UITableViewCellStyleDefault, null) {
-  init {
-    addSubview(view)
+  fun setWidget(widget: UIView) {
+    contentView.addSubview(widget)
+    this.widget = widget
```

Comment:
The use of "widget" here makes me think it's a `Widget` but it's a `UIView`. Can we call it "view" or "child" or something?